src/compiler/checker.ts

```diff
@@ -13523,14 +13523,34 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return instantiateType(instantiable, createTypeMapper([type.indexType, type.objectType], [getNumberLiteralType(0), createTupleType([replacement])]));
     }
 
+    function getLimitedConstraint(type: ReverseMappedType) {

```

Comment:
I'd really like a comment explaining a "limited constraint" is, maybe with an example or two. Otherwise LGTM